loopy/symbolic.py

```diff
@@ -224,30 +241,31 @@ def map_sub_array_ref(self, expr: SubArrayRef,
 
         return SubArrayRef(cast("tuple[p.Variable, ...]", new_inames), new_subscript)
 
-    def map_resolved_function(self, expr, *args: P.args, **kwargs: P.kwargs):
+    def map_resolved_function(self, expr: ResolvedFunction,
+                              *args: P.args, **kwargs: P.kwargs) -> Expression:
         # leaf, doesn't change
         return expr
 
-    def map_type_cast(
-                self, expr: TypeCast,
-                *args: P.args, **kwargs: P.kwargs
-            ) -> Expression:
+    def map_type_cast(self, expr: TypeCast,
+                      *args: P.args, **kwargs: P.kwargs) -> Expression:
         new_child = self.rec(expr.child, *args, **kwargs)
 
         if new_child is expr.child:
             return expr
 
         return type(expr)(expr.type, new_child)
 
-    map_linear_subscript = IdentityMapperBase.map_subscript
-
-    map_rule_argument = map_group_hw_index
+    map_linear_subscript: Callable[Concatenate[Self, LinearSubscript, P],
+                                   Expression] = IdentityMapperBase.map_subscript
```

Comment:
`pyright` is still upset about these.. for mostly obvious reasons :(

Reply:
:shrug: So be it.

The `map_rule_argument` complaint just below can be fixed though, by adding `| RuleArgument` to `map_group_hw_index`.